File: src/main/java/com/theonlytails/ruby/registries/ContainerTypeRegistry.java
package com.theonlytails.ruby.registries;

import com.theonlytails.ruby.RubyMod;
import com.theonlytails.ruby.containers.RubyBarrelContainer;
import net.minecraft.inventory.container.ContainerType;
import net.minecraftforge.common.extensions.IForgeContainerType;
import net.minecraftforge.fml.RegistryObject;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;

public class ContainerTypeRegistry {
    public static final DeferredRegister<ContainerType<?>> CONTAINER_TYPES = DeferredRegister.create(ForgeRegistries.CONTAINERS, RubyMod.MOD_ID);

    public static final RegistryObject<ContainerType<RubyBarrelContainer>> RUBY_BARREL = CONTAINER_TYPES.register(
            "ruby_barrel",
            () -> IForgeContainerType.create(RubyBarrelContainer::new)
    );
}

File: src/main/kotlin/com/theonlytails/rubymod/registries/ContainerTypeRegistry.kt
package com.theonlytails.rubymod.registries

import com.theonlytails.rubymod.MOD_ID
import com.theonlytails.rubymod.containers.RubyBarrelContainer
import net.minecraft.inventory.container.ContainerType
import net.minecraftforge.common.extensions.IForgeContainerType
import net.minecraftforge.registries.ForgeRegistries
import thedarkcolour.kotlinforforge.forge.KDeferredRegister

/**
 * Registers custom containers.
 *
 * @author TheOnlyTails
 */
object ContainerTypeRegistry {
	val containerTypes = KDeferredRegister(ForgeRegistries.CONTAINERS, MOD_ID)

	val rubyBarrel: ContainerType<RubyBarrelContainer> by containerTypes.registerObject("ruby_barrel") {
		IForgeContainerType.create { windowId, playerInventory, data ->
			RubyBarrelContainer(windowId,
				playerInventory,
				data)
		}
	}
}
